src/components/UpdatePaymentDetailsPopup/UpdatePaymentDetailsPopup.js

```diff
@@ -98,6 +100,9 @@ const UpdatePaymentDetailsPopup = ({
         }
       });
     }
+    return () => {
+      dispatch(resetPaymentDetailsPopupState());
```

Comment:
regarding above comment -> probably this fixes the case when user immediately opens the popup again